shared/src/main/scala/effekt/Typer.scala

```diff
@@ -571,7 +579,13 @@ class Typer extends Phase[Module, Module] { typer =>
       effs = (effs ++ (stmtEffs -- handled))
     }
 
-    (params zip args) foreach { case (ps, as) => checkArgumentSection(ps, as) }
+    (params zip args) foreach {
+      // Special case - treat 1st param type () => T as T
```

Comment:
Maybe you could expand a bit on this comment (and maybe add an example?). This part is really the most tricky bit, I believe.

Reply:
I reconsidered about it, it is unnecessary. So I removed it.